File: level 2/연습문제/N개의 최소공배수(유클리드 호제법)/solution.py
def gcd(num1, num2):
    while True:
        if num1 > num2:
            num1 -= num2
        elif num2 > num1:
            num2 -= num1
        else:
            return num1
        
def lcm(num1, num2):
    return num1 * num2 / gcd(num1, num2)

def solution(arr):
    for i in range(len(arr)-1):
        arr[i+1] = lcm(arr[i], arr[i+1])
    return arr[-1]

File: level 2/연습문제/N개의 최소공배수(유클리드 호제법)/solution.js
function gcd(x, y) {
  while (x != y) {
    x = x > y ? x - y : x;
    y = y > x ? y - x : y;
  }
  return x;
}

function solution(arr) {
  return arr.reduce((acc, cur) => (acc * cur) / gcd(acc, cur), 1);
}
